service/history/replicatorQueueProcessor.go

```diff
@@ -38,26 +39,34 @@ import (
 	replicationspb "go.temporal.io/server/api/replication/v1"
 	"go.temporal.io/server/common"
 	"go.temporal.io/server/common/backoff"
+	"go.temporal.io/server/common/convert"
 	"go.temporal.io/server/common/log"
 	"go.temporal.io/server/common/log/tag"
 	"go.temporal.io/server/common/metrics"
 	"go.temporal.io/server/common/persistence"
 	"go.temporal.io/server/common/persistence/versionhistory"
+	"go.temporal.io/server/common/primitives/timestamp"
+	"go.temporal.io/server/service/history/configs"
 	"go.temporal.io/server/service/history/shard"
 )
 
 type (
 	replicatorQueueProcessorImpl struct {
 		currentClusterName string
 		shard              shard.Context
+		config             *configs.Config
 		historyCache       *historyCache
 		executionMgr       persistence.ExecutionManager
 		historyMgr         persistence.HistoryManager
 		metricsClient      metrics.Client
 		logger             log.Logger
 		retryPolicy        backoff.RetryPolicy
-		// This is the batch size used by pull based RPC replicator.
-		fetchTasksBatchSize int
+		pageSize           int
+
+		sync.Mutex
```

Comment:
I think it is good practice to place embedded types on top but in this case, I believe it worth to be a named field `maxTaskIDMutex`.